src/TodoContainer.js

```diff
@@ -0,0 +1,264 @@
+        fields: {
+          title: newTodo.title,
+          deadline: newTodo.deadline,
+        },
+      };
+
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+        },
+        body: JSON.stringify(airtableData),
+      };
+
+      const response = await fetch(
+        `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`,
+        options
+      );
+
+      if (!response.ok) {
+        throw new Error(`The following error has occurred: ${response.status}`);
+      }
+
+      const dataResponse = await response.json();
+
+      setTodoList([
+        ...todoList,
+        {
+          title: newTodo.title,
+          deadline: newTodo.deadline,
+          id: dataResponse.id,
+        },
+      ]);
+
+      return dataResponse;
+    } catch (error) {
+      console.log(error.message);
+      return null;
+    }
+  };
+
+  const removeTodo = async (id) => {
+    try {
+      const newTodoList = todoList.filter((todoItem) => id !== todoItem.id);
+      setTodoList(newTodoList);
+
+      const options = {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+        },
+      };
+
+      const response = await fetch(
+        `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/${id}`,
+        options
+      );
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const editTodo = async (updatedTodo) => {
+    try {
+      console.log(updatedTodo);
+      const airtableData = {
+        fields: {
+          title: updatedTodo.title,
+          deadline: updatedTodo.deadline,
+          completed: updatedTodo.completed,
+        },
+      };
+
+      const options = {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+        },
+        body: JSON.stringify(airtableData),
+      };
+
+      const response = await fetch(
+        `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/${updatedTodo.id}`,
+        options
+      );
+      const dataResponse = await response.json();
+      fetchData();
+
+      return dataResponse;
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const sortTasks = (list, sortType, listName) => {
+    if (sortType === "A-Z ascending") {
+      const sortedList = list.sort((objectA, objectB) => {
+        if (objectA.title < objectB.title) {
+          return -1;
+        }
+        if (objectA.title > objectB.title) {
+          return 1;
+        }
+
+        return 0;
```

Comment:
Could we make this simpler by saying `return objectA.title <= objectB.title ? -1 : 1;`? If the titles are equal, it doesn't matter which comes first so it can lump in with the case where A comes before B, returning -1 and putting A first.